### PR TITLE
Moving logic to only create the drop in the end

### PR DESCRIPTION
**Pull request by Plucks77**

### CODE REVIEW

File: backend/app/controllers/nft_controller.rb [hunk 1]
```diff
@@ -29,91 +27,113 @@ def upload_excel
       i += 2
```
Comment (fc-anjos) [inline backend/app/controllers/nft_controller.rb:27]: Are we iterating every other row? Look into `.each_slice(2)`.

Anyway, this tabs iteration could be its own method, there's a lot of logic going on here. Extracting what should be done for each nft in a method, filtering first or iterating over what's needed will be cleaner

File: backend/app/models/nft.rb [hunk 2]
```diff
@@ -39,8 +37,29 @@ def self.import_from_hash(hash, _drop_name = nil, drop_id)
...
+      :award,
+      :price,
+      :gallery_filename,
+      :final_filename,
+      :currency,
+      :gallery_url,
+      :final_type,
+      :gallery_type
+    )
     pp 'FIELDS', fields
```
Comment (fc-anjos) [inline backend/app/models/nft.rb:63]: remove pp

File: backend/app/models/nft.rb [hunk 3]
```diff
@@ -49,34 +68,41 @@ def self.import_from_hash(hash, _drop_name = nil, drop_id)
...
   end
 
   def has_watermark?
-    File.exists?(self.watermark_filename)
+    File.exist?(watermark_filename)
   end
-  
-  def do_watermark(src, dst, watermark="./app/assets/images/watermark-45.png")
+
+  def do_watermark(src, dst, watermark = './app/assets/images/watermark-45.png')
```
Comment (fc-anjos) [inline backend/app/models/nft.rb:96]: "Do" and "Make" watermark seem like confusing names

**Review by fc-anjos — CHANGES_REQUESTED** [review]

**Comment by fc-anjos** [comment]
Hey @umschaudhary,

there are a bunch of additions here that seem like linter changes:
Switching " to ' and adding ;
or:

```
-    File.exists?(self.watermark_filename)
+    File.exist?(watermark_filename)
```

We all need to connect on this, make sure we're using the same linter configuration, open a single PR to lint it all and not having it spread across a lot of small changes. Makes it much harder to review.
@Plucks77 @thenifemi @jonathastavares

File: backend/app/models/nft.rb (hunk 3 above)
Comment (umschaudhary) [reply backend/app/models/nft.rb]: These are commented out since not used anywhere

File: backend/app/controllers/nft_controller.rb (hunk 1 above)
Comment (umschaudhary) [reply backend/app/controllers/nft_controller.rb]: segregated logic to seperate methods

File: backend/app/controllers/nft_controller.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class NftController < ApplicationController
   require 'roo'
-
+  require 'securerandom'
```
[the remark this answers is not in the document]
Comment (umschaudhary) [reply backend/app/controllers/nft_controller.rb]: removed

File: backend/app/services/nft_service.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    def parse_spreadsheet(xlsx)
+      tabs = xlsx.sheets
+      random_string = SecureRandom.hex
+      nfts = []
+      tabs.each do |tab|
+        sheet = xlsx.sheet(tab)
+        last = sheet.last_row + 1
+
+        headers = sheet.row(6)
+        (8..last).each do |i| # 8th row is a starting row
```
Comment (fc-anjos) [inline backend/app/services/nft_service.rb:14]: Almost there! Let's filter for nil first, and then map the nfts. We don't need this empty array as a sidecar

**Review by fc-anjos — CHANGES_REQUESTED** [review]

**Review by fc-anjos — APPROVED** [review]

**Pull request merged**